modules/build/src/main/scala/scala/build/errors/BuildException.scala

```diff
@@ -2,6 +2,12 @@ package scala.build.errors
 
 import scala.build.Position
 
+case class Diagnostic(
+  val message: String,
+  val positions: Seq[Position] = Nil,
+  val severity: Severity
+)
```

Comment:
NIT No need to put `val` in case classes, and it's more convenient to put the fields with default values at the end:
```suggestion
case class Diagnostic(
  message: String,
  severity: Severity,
  positions: Seq[Position] = Nil
)
```